openstack/identity/v3/users/requests.go

```diff
@@ -49,11 +50,22 @@ type ListOpts struct {
 
 	// UniqueID filters the response by unique ID.
 	UniqueID string `q:"unique_id"`
+
+	// Filters filters the response by custom filters such as
+	// 'name__contains=foo'
+	Filters map[string]string `q:"-"`
 }
 
 // ToUserListQuery formats a ListOpts into a query string.
 func (opts ListOpts) ToUserListQuery() (string, error) {
 	q, err := gophercloud.BuildQueryString(opts)
+
+	params := q.Query()
+	for k, v := range opts.Filters {
+		params.Add(k, v)
```

Comment:
At a minimum, the filter must have two underscores, so maybe we should do a string check to make sure each key has two underscores? Otherwise `Filter` could be used to pass any arbitrary key into the query. Thoughts?